src/main/cpp/src/parse_uri.cu

```diff
@@ -495,7 +495,7 @@ bool __device__ validate_fragment(string_view fragment)
 __device__ std::pair<string_view, bool> find_query_part(string_view haystack, string_view needle)
 {
   auto const n_bytes     = needle.size_bytes();
-  auto const find_length = haystack.size_bytes() - n_bytes + 1;
+  auto const find_length = haystack.size_bytes() - n_bytes;
 
   auto h           = haystack.data();
   auto const end_h = haystack.data() + find_length;
```

Comment:
Be careful, line 517 line 520 have conflict IMO:
```
517: h++;
520: h++;
```

517 skip some bytes and p will point to next entry key=value.
But 520 will add h pointer again, then IMO the logic is not what we expect.

Reply:
line 520 is used to skip the `&`, and `*h != '='`  in line 524 is short-circuited if h is really out of range, so it looks correct. but the logic is confusing, will try to refactor it.